src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/GeneExpressionAnnotationFmsDTOValidator.java

```diff
@@ -212,6 +212,12 @@ private ObjectResponse<AnatomicalSite> validateAnatomicalSite(GeneExpressionFmsD
 				if (cellularComponent == null) {
 					response.addErrorMessage("whereExpressed - cellularComponentTermId", ValidationConstants.INVALID_MESSAGE + " (" + geneExpressionFmsDTO.getWhereExpressed().getCellularComponentTermId() + ")");
 				} else {
+					GOTerm cellularComponentRibbon = goTermService.findSubsetTerm(cellularComponent, "goslim_agr");
+					if (cellularComponentRibbon == null) {
+						anatomicalSite.setCellularComponentOther(true);
+					} else {
+						anatomicalSite.setCellularComponentRibbonTerm(cellularComponentRibbon);
```

Comment:
I think you also need to set cellularComponentOther to false here, otherwise if it's an update where the value was previously true then it would remain as true.

Reply:
Sure, that makes sense. I can add that.

Reply:
@adamgibs , do we have all the GoTerms in the  "goslim_agr" subset already loaded?

Reply:
@abecerra Yeah, as far as I know. There are GOTerms with subsets that include "goslim_agr" currently.